staging/src/k8s.io/apiserver/pkg/admission/reinvocation/reinvocation.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reinvocation
+
+import (
+	"k8s.io/apiserver/pkg/admission"
+)
+
+// WithReinvocation creates a handler that wraps the provided handler and reinvokes it according
+// if needed according to reinvocation policy of the webhooks.
+func WithReinvocation(wrapped admission.Interface) admission.Interface {
```

Comment:
not sure about the usefulness of this wrapper. It only applies to admission chains in a sensible way, doesn't it?

Reply:
would just put that into k8s.io/apiserver/pkg/server/options/admission.go and keep it private.

Reply:
Yes, it's not fully compose-able. I'll make it private and document it for clarity.